src/contracts/pods/EigenPod.sol

```diff
@@ -347,18 +288,69 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
         eigenPodManager.recordBeaconChainETHBalanceUpdate(podOwner, int256(totalAmountToBeRestakedWei));
     }
 
-    /// @notice Called by the pod owner to withdraw the nonBeaconChainETHBalanceWei
-    function withdrawNonBeaconChainETHBalanceWei(
-        address recipient,
-        uint256 amountToWithdraw
-    ) external onlyEigenPodOwner onlyWhenNotPaused(PAUSED_NON_PROOF_WITHDRAWALS) {
+    /**
+     * @dev Prove that one or more validators were slashed on the beacon chain and have not had timely
+     * checkpoint proofs since being slashed. If successful, this allows the caller to start a checkpoint.
+     * @dev Note that in order to start a checkpoint, any existing checkpoint must already be completed!
+     * (See `_startCheckpoint` for details)
+     * @param beaconTimestamp the beacon chain timestamp sent to the 4788 oracle contract. Corresponds
+     * to the parent beacon block root against which the proof is verified.
+     * @param stateRootProof proves a beacon state root against a beacon block root
+     * @param proof the fields of the beacon chain "Validator" container, along with a merkle proof against
+     * the beacon state root. See the consensus specs for more details:
+     * https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#validator
+     *
+     * @dev Staleness conditions:
+     * - Validator's last balance update is older than `beaconTimestamp` by `TIME_TILL_STALE_BALANCE`
+     * - Validator MUST be in `ACTIVE` status in the pod
+     * - Validator MUST be slashed on the beacon chain
+     */
+    function verifyStaleBalance(
```

Comment:
I'm trying to figure out if this is enough conditions. should the caller also prove that the current balance is lower than the stored one?
I am trying to consider the inconvenience to a user when they have many validators pointed to a pod but only one validator is slashed -- it feels like the current logic would essentially force such a user to either exit the slashed validator or do a checkpoint every 2 weeks (as long as someone calls this function)? perhaps that's OK UX tho

Reply:
> should the caller also prove that the current balance is lower than the stored one?

well, one difficult thing here is that the method currently has access to effective balance, not current balance. we could add more to the method, but i'm not sure if the complexity is worth it